presto-main/src/main/java/com/facebook/presto/server/security/SpnegoFilter.java

```diff
@@ -211,9 +212,13 @@ public Principal getUserPrincipal()
         return Optional.empty();
     }
 
-    private static void sendChallenge(HttpServletResponse response, boolean includeRealm, String invalidSpnegoToken)
+    private static void sendChallenge(HttpServletRequest request, HttpServletResponse response, boolean includeRealm, String invalidSpnegoToken)
         throws IOException
     {
+        long contentLength = request.getContentLengthLong();
```

Comment:
They don't want to fix it, claiming that it is not an issue at all. Here is the link: eclipse/jetty.project#938. 

The main problem is that queries are sporadically failing if the query body is large (tested with 32kb query) and kerberos. Server sends 401 UNAUTHORIZED before consuming the content. And that causes request-failure in Jetty client. In case of request failure they don't resend the authenticated request.